### ISSUE TITLE
segfault with TTRSS

### ISSUE THREAD

**Issue by Stunkymonkey** [jangernert/FeedReader#450]
I just expirienced a segmentation fault:

> [ INFO  ] daemon: sync started
> [ DEBUG ] TTRSS: ping
> [ DEBUG ] TTRSS Message: ping successfull
> [WARNING] dbDaemon: Deleting nonexisting categories
> [WARNING] dbDaemon: Deleting articles without feed
> [WARNING] dbDaemon: Deleting unsibscribed feeds
> [ DEBUG ] daemon: symbolicIcon
> [ DEBUG ] feedserver: symbolicIcon
> [ DEBUG ] ttrss URL: https://feeds.stunkymonkey.de/api/
> [ INFO  ] There are 2 unread articles
> [ ERROR ] ttrss api error
> [ DEBUG ] {"sid":"soi5rmf3f6gbtnahvs5nt60qn1","op":"getCompactHeadlines","feed_id":-4,"limit":4000,"view_mode":"unread"}
> [ DEBUG ] {"seq":0,"status":1,"content":{"error":"UNKNOWN_METHOD","method":"getcompactheadlines"}}
> [ ERROR ] ttrss api error
> [ DEBUG ] {"sid":"soi5rmf3f6gbtnahvs5nt60qn1","op":"getCompactHeadlines","feed_id":-4,"limit":4000,"view_mode":"unread"}
> [ DEBUG ] {"seq":0,"status":1,"content":{"error":"UNKNOWN_METHOD","method":"getcompactheadlines"}}
> [ DEBUG ] {"sid":"soi5rmf3f6gbtnahvs5nt60qn1","op":"getCompactHeadlines","feed_id":-4,"limit":4000,"view_mode":"unread"}
> [ DEBUG ] {"sid":"soi5rmf3f6gbtnahvs5nt60qn1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":200,"view_mode":"all_articles"}
> [ DEBUG ] {"sid":"soi5rmf3f6gbtnahvs5nt60qn1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":0,"view_mode":"all_articles"}
> [ DEBUG ] {"sid":"soi5rmf3f6gbtnahvs5nt60qn1","op":"getArticle","article_id":"230132,230131,230130,230129,230125,230126,230128,230127,230124,230123"}
> [ ERROR ] dbErrorLog: 21: misuse at line 361 of [424a0d3803]
> [ DEBUG ] Opening Database: /home/felix/.local/share/feedreader/data/feedreader-04.db
> [ DEBUG ] Utils: generate preview for article: 230123
> [ DEBUG ] Utils: generate preview for article: 230124
> [ DEBUG ] no html to create preview from
> [ DEBUG ] no html to create preview from
> [ DEBUG ] no html to create preview from
> [ DEBUG ] no html to create preview from
> [ DEBUG ] Utils: generate preview for article: 230129
> [ DEBUG ] Utils: generate preview for article: 230130
> [ DEBUG ] Utils: generate preview for article: 230131
> [ DEBUG ] Utils: generate preview for article: 230132
> [ DEBUG ] FeedServer: new articles: 10
> [ DEBUG ] SELECT articleID, feedID FROM main.articles WHERE date <= datetime('now', '-28 days') AND marked = 10 AND rowid BETWEEN 1 AND 1696 
> [ INFO  ] Deleting article "228385"
> [WARNING] DELETING rowID: 928 from  table: articles and db: main
> [ INFO  ] Deleting article "228419"
> [WARNING] DELETING rowID: 930 from  table: articles and db: main
> [ ERROR ] Utils - remove_directory: Error opening directory '/home/felix/.local/share/feedreader/data/images/107/228419': No such file or directory
> [ DEBUG ] daemon: update badge count 2
> [ DEBUG ] FeedServer: grabContent
> [ DEBUG ] Grabber: process article: https://freakshow.fm/fs196-nickt-nicht
> [ DEBUG ] Grabber: no config (freakshow.fm.txt) found for article: https://freakshow.fm/fs196-nickt-nicht
> [ DEBUG ] Grabber: no config (freakshow.fm.txt) - cutSubdomain - found for article: https://freakshow.fm/fs196-nickt-nicht
> [ DEBUG ] grabberUtils: fixIframeSize
> [ DEBUG ] GrabberUtils: repairURL xpath:"//img" attr:"src"
> [ DEBUG ] GrabberUtils: repairURL xpath:"//iframe" attr:"src"
> [ DEBUG ] addAttributes: //a  target _blank
> [ DEBUG ] GrabberUtils: save Images: 229972, 7
> [ DEBUG ] GrabberUtils: download Image https://freakshow.fm/wp-content/cache/podlove/47/08928e3c26dcb1141d67ad75869619/tim-pritlove_50x50.jpg
> [ DEBUG ] Grabber: checkParent
> [ DEBUG ] Grabber: parent td
> [ DEBUG ] GrabberUtils: download Image https://freakshow.fm/wp-content/cache/podlove/06/98a8013f2dd0b64b31d9a37991d4dd/twitter_20x20.png
> [ DEBUG ] Grabber: checkParent
> [ DEBUG ] Grabber: parent a
> [ DEBUG ] Grabber: url https://twitter.com/timpritlove
> [ DEBUG ] Grabber: type text/html
> [ DEBUG ] GrabberUtils: download Image https://freakshow.fm/wp-content/cache/podlove/a4/26976e5d55fca60c010bdf4a44fcc4/website_20x20.png
> [ DEBUG ] Grabber: checkParent
> [ DEBUG ] Grabber: parent a
> [ DEBUG ] Grabber: url http://metaebene.me
> [ DEBUG ] Grabber: type text/html
> [ DEBUG ] GrabberUtils: download Image https://freakshow.fm/wp-content/cache/podlove/a7/de42e214d031f5a10cb2dd670fb8cb/bitcoin_20x20.png
> [ DEBUG ] Grabber: checkParent
> [ DEBUG ] Grabber: parent a
> [ DEBUG ] Grabber: url bitcoin:19unDNWVNsChMpjfhDYYBsrCi2SmBgyKRV
> 
> (feedreader-daemon:29017): libsoup-CRITICAL **: soup_session_send_message: assertion 'SOUP_IS_MESSAGE (msg)' failed
> 
> [1]+  Segmentation fault      (core dumped) /usr/bin/feedreader-daemon

This is the output of the `feedreader-daemon`

Using:

- Arch-Linux
- FeedReader 2.0 installed via Aur
- Tiny Tiny RSS v16.8 (73c4e7d)
- api_feedreader 1.00

**Comment by jangernert** [comment]
The segfault is not caused by the tt-rss backend or anything related. It is caused by this article-URL "bitcoin:19unDNWVNsChMpjfhDYYBsrCi2SmBgyKRV" that the grabber tries to download. Until now I only checked for empty strings and null. But I guess there is some weird stuff out there :P

**Comment by jangernert** [comment]
Btw 2 things:
1. you can work around this issue by deactivating the contant-grabber in the settings
2. the feedreader package of aur is outdated (latest version is 2.0.2). I don't know when or if it will be updated. So I would suggest using feedreader-git for now. This will also include the fix for this report. (Don't forget to restart the daemon after upgrading to make sure the new version is running ;) )

**Comment by Stunkymonkey** [comment]
i updated to `feedreader-git`
but still the same issue

**Comment by jangernert** [comment]
Is your version newer than commit 68f2358ade8e4e99fb33eb98fb4f74fec7b2e71c?
You can check with --version.
Did you kill the running instance of the feedreader-daemon before testing again?
Does the log of the daemon still end with something similar
```
[ DEBUG ] Grabber: url bitcoin:19unDNWVNsChMpjfhDYYBsrCi2SmBgyKRV

(feedreader-daemon:29017): libsoup-CRITICAL **: soup_session_send_message: assertion 'SOUP_IS_MESSAGE (msg)' failed
```

If your answer to all the questions above is yes, can you please provide a backtrace of the crash?
Did you also try to disable the content-grabber in the settings to avoid the crash?

**Comment by Stunkymonkey** [comment]
```$ feedreader --version
Version: 2.0.2
Git Commit: f61d27dd910c51ca5be2d0f8b4065e795ac8e3d3
```
so yes

yes i did kill the feedreader-daemon and started it again

yes the log still looks like this

yes i tried to disable the content grabber, but it still crashed

about the backtrace: what kind of logs do you need
because all of the output i get of the feedreader-daemon is already posted above

**Comment by jangernert** [comment]
Okay, I could actually reproduce the crash with a bitcoin-url like yours. Seems this is a valid uri to soup, but the message is still null.
I'm pretty sure it should be fixed with e50c60fa0e22c8f59766ac3990eb44d1533ae59f now :D 
Can you please test again? Just update and restart the daemon ;)

**Comment by Stunkymonkey** [comment]
i build it again and started everything again:
```
[ DEBUG ] GrabberUtils: download Image https://freakshow.fm/wp-content/cache/podlove/a4/26976e5d55fca60c010bdf4a44fcc4/website_20x20.png
[ DEBUG ] Grabber: checkParent
[ ERROR ] error writing log to file: Stream has outstanding operation
[ ERROR ] [ DEBUG ] Grabber: checkParent

[ DEBUG ] Grabber: parent a
[ DEBUG ] Grabber: url http://metaebene.me
[ DEBUG ] Grabber: type text/html
[ DEBUG ] GrabberUtils: download Image https://freakshow.fm/wp-content/cache/podlove/a7/de42e214d031f5a10cb2dd670fb8cb/bitcoin_20x20.png
[ DEBUG ] Grabber: checkParent
[ DEBUG ] Grabber: parent a
[ DEBUG ] Grabber: url bitcoin:19unDNWVNsChMpjfhDYYBsrCi2SmBgyKRV

(feedreader-daemon:6844): libsoup-CRITICAL **: soup_session_send_message: assertion 'SOUP_IS_MESSAGE (msg)' failed

[1]+  Segmentation fault      (core dumped) /usr/bin/feedreader-daemon
[felix:~] $ feedreader --version
Version: 2.0.2
Git Commit: e50c60fa0e22c8f59766ac3990eb44d1533ae59f
```
still the same

**Comment by jangernert** [comment]
No not the same issue. Now the bitcoin-url crashes another soup.mesage :P
Btw: why is the url of the "article" not a valid website-url?

**Comment by jangernert** [comment]
Added checks to the code that is crashing for you and some more soup messages da88ce67b36ead0f2b648e02ab2bbbc2c02f4eab

**Comment by Stunkymonkey** [comment]
this is the feed is subscribed to:

> http://mobilemacs.de/feed

**Comment by Stunkymonkey** [comment]
error when compiling:
```
[  4%] Generating src/Logger.c;src/Enums.c;src/dbBase.c;src/Settings.c;src/Structs.c;src/Utils.c;src/QueryBuilder.c;src/ContentGrabber/grabberUtils.c;src/ContentGrabber/stringPair.c;src/Model/Feed.c;src/Model/Category.c;src/Model/Article.c;src/Model/Tag.c;src/Model/CachedAction.c;src/Model/InterfaceState.c;src/Model/ShareAccount.c;Constants.c
~/.cache/pacaur/feedreader-git/src/FeedReader/src/ContentGrabber/grabberUtils.vala:533.16-533.18: error: The name `msg' does not exist in the context of `FeedReader.grabberUtils.downloadImage'
            if(msg == null)
Compilation failed: 1 error(s), 27 warning(s)
make[2]: *** [CMakeFiles/FeedReaderCommon.dir/build.make:130: libFeedReader_valac.stamp] Error 1
make[1]: *** [CMakeFiles/Makefile2:67: CMakeFiles/FeedReaderCommon.dir/all] Error 2
make: *** [Makefile:128: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
:: failed to build feedreader-git package(s)
```

**Comment by jangernert** [comment]
Yes, saw that as well just now. It's pretty late =/ Fixed in master now
I also subscribed to the feed and it is grabbing content without crashing now (progress 8/15)

**Comment by Stunkymonkey** [comment]
thanks now it works

sorry for the quirky bug-report

very nice work you are doing there keep it up!

**Comment by thedailyplanet** [comment]
A similar problem here:

`[ DEBUG ] TTRSS: ping
[ DEBUG ] TTRSS Message: ping successfull
[WARNING] dbDaemon: Deleting nonexisting categories
[WARNING] dbDaemon: Deleting articles without feed
[WARNING] dbDaemon: Deleting unsibscribed feeds
[ DEBUG ] daemon: symbolicIcon
[ DEBUG ] feedserver: symbolicIcon
[ DEBUG ] ttrss URL: https://upsetter.xen.prgmr.com/ttrss/api/
[ INFO  ] There are 4019 unread articles
[ ERROR ] ttrss api error
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getCompactHeadlines","feed_id":-4,"limit":50000,"view_mode":"unread"}
[ DEBUG ] {"seq":0,"status":1,"content":{"error":"UNKNOWN_METHOD","method":"getcompactheadlines"}}
[ ERROR ] ttrss api error
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getCompactHeadlines","feed_id":-4,"limit":50000,"view_mode":"unread"}
[ DEBUG ] {"seq":0,"status":1,"content":{"error":"UNKNOWN_METHOD","method":"getcompactheadlines"}}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getCompactHeadlines","feed_id":-4,"limit":50000,"view_mode":"unread"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":4800,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":4600,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":4400,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":4200,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":4000,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":3800,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":3600,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":3400,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":3200,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":3000,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":2800,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":2600,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":2400,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":2200,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":2000,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":1800,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":1600,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":1400,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":1200,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":1000,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":800,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":600,"view_mode":"all_articles"}
--- SIGWINCH {si_signo=SIGWINCH, si_code=SI_KERNEL} ---
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":400,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":200,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getHeadlines","feed_id":-4,"limit":200,"skip":0,"view_mode":"all_articles"}
[ DEBUG ] {"sid":"47e5pbgnjjj1onapqh6c7fe2j1","op":"getArticle","article_id":"118514,118515,118516,118517,118518,118513,118511,118512,118510,118509,118507,118508,118506,118505,118504,118503,118500,118501,118502,118494,118495,118496,118497,118498,118499,118493,118492,118490,118488,118487,118491,118489"}
[ DEBUG ] Utils: generate preview for article: 118487
[ DEBUG ] Utils: generate preview for article: 118488
[ DEBUG ] Utils: generate preview for article: 118489
[ DEBUG ] Utils: generate preview for article: 118490
[ DEBUG ] Utils: generate preview for article: 118491
[ DEBUG ] Utils: generate preview for article: 118492
[ DEBUG ] Utils: generate preview for article: 118493
[ DEBUG ] Utils: generate preview for article: 118494
[ DEBUG ] Utils: generate preview for article: 118495
[ DEBUG ] Utils: generate preview for article: 118496
[ DEBUG ] Utils: generate preview for article: 118497
[ DEBUG ] Utils: generate preview for article: 118498
[ DEBUG ] Utils: generate preview for article: 118499
[ DEBUG ] Utils: generate preview for article: 118500
[ DEBUG ] Utils: generate preview for article: 118501
[ DEBUG ] Utils: generate preview for article: 118502
[ DEBUG ] Utils: generate preview for article: 118503
[ DEBUG ] Utils: generate preview for article: 118504
[ DEBUG ] Utils: generate preview for article: 118505
[ DEBUG ] no html to create preview from
[ DEBUG ] Utils: generate preview for article: 118507
[ DEBUG ] Utils: generate preview for article: 118508
[ DEBUG ] Utils: generate preview for article: 118509
[ DEBUG ] Utils: generate preview for article: 118510
[ DEBUG ] Utils: generate preview for article: 118511
[ DEBUG ] Utils: generate preview for article: 118512
[ DEBUG ] Utils: generate preview for article: 118513
[ DEBUG ] Utils: generate preview for article: 118514
[ DEBUG ] Utils: generate preview for article: 118515
[ DEBUG ] Utils: generate preview for article: 118516
[ DEBUG ] Utils: generate preview for article: 118517
[ DEBUG ] Utils: generate preview for article: 118518
[ DEBUG ] FeedServer: new articles: 32
[ DEBUG ] FeedServer: grabContent
[ DEBUG ] grabberUtils: fixIframeSize
[ DEBUG ] GrabberUtils: repairURL xpath:"//img" attr:"src"
[ DEBUG ] GrabberUtils: repairURL xpath:"//iframe" attr:"src"
[ DEBUG ] addAttributes: //a  target _blank
[ DEBUG ] GrabberUtils: save Images: 118484, 12
[ DEBUG ] GrabberUtils: download Image https://public.tableau.com/static/images/Da/DashboardBR/Dashboard/1_rss.png
[ DEBUG ] Grabber: checkParent
[ DEBUG ] Grabber: parent a
[ DEBUG ] Grabber: url #

(feedreader-daemon:6031): libsoup-CRITICAL **: soup_session_send_message: assertion 'SOUP_IS_MESSAGE (msg)' failed
+++ killed by SIGSEGV +++
Segmentation fault
`
Looks like a failure to parse the malformed URL extracted from the article ( https://theintercept.com/2017/09/26/mapeamento-inedito-mostra-que-doacoes-legais-da-odebrecht-beneficiaram-1-087-candidatos-desde-2002/ )

I'm running FeedReader 2.0.2, built from flatpak install on Linux Mint 18.

I've a long list of other issues I've had to fix or are still outstanding with this installation. 2 days in and I've still to read an article in the UI, but that's another story..

**Comment by jangernert** [comment]
The issue is already fixed in master branch. Basically creating a Soup.Message with the provided URL `#` leads to an invalid message which crashes FeedReader if I try to send it.
I'm still waiting on the meson port of the buildsystem by @bil-elmoussaoui to release an update. Hopefully it won't be long ;)
